### DISCUSSION TITLE
Use annotationType instead of getClass

### DISCUSSION THREAD

**Discussion by charlescrookes** [smallrye/smallrye-graphql#1241] [General]
Access type of annotation directly, not through the getClass() that creates a proxy. This allows for the library to be used seamlessly in GraalVM Native applications.

https://github.com/smallrye/smallrye-graphql/blob/2c48d04f251b3a39de4344d985e1b75e3d78e194/client/implementation/src/main/java/io/smallrye/graphql/client/impl/typesafe/reflection/FieldInfo.java#L71

**Comment by jmartisk** [comment]
Hi, feel free to submit a PR ;)
In any case, this piece of code will probably change soon anyway, because I'm planning to reduce the amount of reflection done at runtime. Annotation value is something that you can scan once during build.

**Comment by charlescrookes** [comment]
Good afternoon,

Created PR https://github.com/smallrye/smallrye-graphql/pull/1243 for this change.  Verified change locally with `io.quarkus:quarkus-smallrye-graphql-client:2.6.1.Final` using `com.fasterxml.jackson.annotation.JsonProperty` annotation.

Thanks,
Charles Crookes
